neuralpredictors/training/tracking.py

```diff
@@ -1,6 +1,15 @@
-from collections import defaultdict
-import numpy as np
+import copy
 import time
+import logging
+from collections import defaultdict, abc
+from typing import Dict, Tuple, List, Mapping, Union, Optional, Sequence
+import numpy as np
+from tqdm import tqdm
+from __future__ import annotations
```

Comment:
This needs to be the first line of code in the file (i.e. right after the docstring). That is why the tests are failing.